src/main/java/org/opentripplanner/street/model/edge/StreetVehicleParkingLink.java

```diff
@@ -68,8 +68,10 @@ public State[] traverse(State s0) {
     }
 
     var vehicleParking = vehicleParkingEntranceVertex.getVehicleParking();
-    final VehicleParkingRequest parkingRequest = s0.getRequest().parking();
-    if (traversalBanned(parkingRequest, vehicleParking)) {
+    final VehicleParkingPreferences parkingPreferences = s0.currentMode() == TraverseMode.CAR
+      ? s0.getRequest().preferences().car().parking()
+      : s0.getRequest().preferences().bike().parking();
```

Comment:
Can you create a utility method for this somewhere in the preferences?

Reply:
I created a method under the root preferences.

Reply:
As I mentioned on todays meeting, we are passing in the API request/preferences into AStar without a mapping it. One of the effects is that we then need to do mapping everywhere it is used instead of in one place - I do not expect you tro refactor this, but I just wanted to highlight the pain. If it is interesting, we could take one feature like the parking and refactor AStar with just that. Then the job is not so big, and it can be done in iterations. Also, when a pattern is established it become easier to take do the rest - just follow the recipe.